datadog/fakeintake/fakeintake.go

```diff
@@ -0,0 +1,58 @@
+package fakeintake
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	stackKey = "fakeintake-url"
+)
+
+// PulumiData pulumi side data
+type PulumiData struct {
+	Url pulumi.StringInput
+}
+
+type exporter struct {
+	stackKey string
+}
+
+// ClientData client side data
+type ClientData struct {
+	Url string
+}
+
+// GetClientDataDeserializer
+func (exporter *exporter) GetClientDataDeserializer() func(auto.UpResult) (*ClientData, error) {
+	return func(result auto.UpResult) (*ClientData, error) {
+		outputs, found := result.Outputs[exporter.stackKey]
+		if !found {
+			return nil, fmt.Errorf("cannot find %v in the stack result", exporter.stackKey)
+		}
+		url, ok := outputs.Value.(string)
+		if !ok {
+			return nil, fmt.Errorf("the type %v is not valid for the key %v", reflect.TypeOf(outputs.Value), exporter.stackKey)
+		}
+		return &ClientData{Url: url}, nil
+	}
+}
+
+type PulumiExporter struct {
```

Comment:
I know we discussed this topic in person but when looking at the code I think it would be easier to understand if:
 - `PulumiExporter` is defined in the same file as its constructor `NewEcsFakeintake`
 - Is there a reason to named the struct `PulumiExporter` and not something like `FakeIntake`?
 - Because there are two fields in `PulumiExporter` (`Url` and `stackKey`), I think having only one struct is enough instead of having 3 (`PulumiExporter`, `exporter` and `PulumiData`)

Reply:
Names can be improved, I wanted to split the three parts that are here:

* the exporter, that requires as stack key, and that know where to serialise and where to deserialise
* the data pulumi side, where `url` is a `pulumi.StringOutput`
* the data on the runtime side, where `url` is a `string`

For instance, the deserialise function does not need the pulumi data. This is why it is the `exporter` that has a `Deserialise` function.

On the other side the export action does not need to know about the runtime. I find having everything in the same struct confusing, but I agree names are not explicit enough.